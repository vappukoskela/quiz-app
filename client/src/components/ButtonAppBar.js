
import React, {useState} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { Button, IconButton, MenuItem, Select } from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import strings from '../localization/strings';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  lanselect: {
    margin: theme.spacing(1),
    width: "70px",
    paddingInlineStart: '15px'
  }
}));


 function ButtonAppBar(props) {
   console.log(props.language)
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="fixed">
        <Toolbar>
            <Typography variant="h6" className={classes.title}>
            <Link to="/" style={{ textDecoration: 'none', color: 'white' }}>
              Quiz App
            </Link>
          </Typography>
            <Link to="/login" style={{ textDecoration: 'none' }}>
              <Button variant="contained" color="primary" disableElevation>{strings.login}</Button>
            </Link>
            {console.log(props, "props")}
            {console.log(strings, "strings")}
            <Select className={classes.lanselect} defaultValue="en" style={{  color: 'white' }} >
              <MenuItem onClick={()=> props.switchLanguage('en')} value="en" selected> EN </MenuItem>
              <MenuItem onClick={()=> props.switchLanguage('fi')} value="fi"> FI </MenuItem>
            </Select>
        </Toolbar>
      </AppBar>

    </div>
  );
}

export default ButtonAppBar
