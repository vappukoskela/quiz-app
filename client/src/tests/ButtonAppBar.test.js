import React from 'react';
import renderer from 'react-test-renderer';
import ButtonAppBar from '../components/ButtonAppBar.js';

it('renders correctly', () => {
    const tree = renderer.create(<ButtonAppBar />).toJSON();
    expect(tree).toMatchSnapshot();
});