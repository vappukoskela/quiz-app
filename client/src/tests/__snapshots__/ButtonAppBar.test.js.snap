// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
      >
        Quiz App
      </h6>
    </div>
  </header>
</div>
`;
